assignment6/test_rearranging_cars.py

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/python3.5
+import unittest
+import rearranging_cars
+
+class TestRearrangingCars(unittest.TestCase):
+    def setUp(self):
+        self.initial = [1, 0, 2]
+
+    def test_no_movements(self):
+        moves = rearranging_cars.rearrange(self.initial, self.initial)
+        self.assertEqual(0, len(moves))
+
+    def test_one_movement(self):
+        final = [1, 2, 0]
+        expected_moves = [rearranging_cars.Movement(2, 2, 1)]
+        actual_moves = rearranging_cars.rearrange(self.initial, final)
+        self.assertMovementListEquals(expected_moves, actual_moves)
+
+    def test_multiple_movements(self):
+        final = [2, 0, 1]
+        expected_moves = [rearranging_cars.Movement(1, 0, 1),
+                          rearranging_cars.Movement(2, 2, 0),
+                          rearranging_cars.Movement(1, 1, 2)]
+        actual_moves = rearranging_cars.rearrange(self.initial, final)
+        self.assertMovementListEquals(expected_moves, actual_moves)
+
+
+    def assertMovementListEquals(self, expected_moves, actual_moves):
```

Comment:
One alternative would be instead of checking the steps as well to check if the final state is achieved correctly :)